packages/react-form/src/hooks/list/test/dynamiclist.test.tsx

```diff
@@ -39,6 +48,26 @@ describe('useDynamicList', () => {
       );
     }
 
+    it('can move field to a new position', () => {
```

Comment:
Would be nice to have a test to verify that we're throwing an exception on invalid indices. (A fully satisfactory test would test all 4 cases, but we can keep it simple and just do one case for now.)